refactor(ivy): remove interpolation instructions

crisbeto opened the pull request:
Makes the `interpolateX` instructions internal-only and removes their use of global state.

This PR resolves FW-1387.

alfaproject commented on packages/core/src/render3/instructions/attribute_interpolation.ts:
```diff
@@ -271,7 +272,7 @@ export function ɵɵattributeInterpolate6(
   const lView = getLView();
   // TODO(FW-1340): Refactor to remove the use of other instructions here.
```
I'm guessing this comment no longer applies (same in other places)

kara replied:
Yep, these comments should be removed

crisbeto commented on packages/core/src/render3/instructions/interpolation.ts:
```diff
@@ -28,14 +26,11 @@ import {storeBindingMetadata} from './shared';
- * @codeGenApi
  */
-export function ɵɵinterpolationV(values: any[]): string|NO_CHANGE {
+export function interpolationV(lView: LView, values: any[]): string|NO_CHANGE {
```
Question for reviewer: should this file stay under `instruction` or be moved some place else? If yes, what would be a better place?

kara replied:
Let's keep it here for now

kara reviewed (approved):
LGTM, let's just clean up the comments in that one file

crisbeto commented:
Removed the comments.

kara commented on packages/core/src/render3/instructions/class_map_interpolation.ts:
```diff
@@ -33,8 +35,7 @@ import {ɵɵclassMap} from './styling';
 export function ɵɵclassMapInterpolate1(prefix: string, v0: any, suffix: string): void {
-  // TODO(FW-1340): Refactor to remove the use of other instructions here.
-  const interpolatedValue = ɵɵinterpolation1(prefix, v0, suffix);
+  const interpolatedValue = interpolation1(getLView(), prefix, v0, suffix);
```
I actually think we need to keep it here since we haven't cleaned up `classMap` yet

crisbeto replied:
Re-added them. FWIW I kept the call through directly to the instruction in my previous pass over `classMap`, because the internal function would've had the exact same signature.

crisbeto committed:
refactor(ivy): remove interpolation instructions

Makes the `interpolateX` instructions internal-only and removes their use of global state.

This PR resolves FW-1387.

kara commented:
[VE presubmit](http://test/OCL:256435696:BASE:256435719:1562188239906:8dcd689d)

[Ivy presubmit](https://test.corp.google.com/ui#id=OCL:256435696:BASE:256437955:1562189039344:2e3bae77)

kara commented:
merge-assistance: global
